packages/itwinui-react/src/core/utils/components/Popover.tsx

```diff
@@ -4,162 +4,271 @@
+    role,
+  } = options;
+
+  const [open, onOpenChange] = useControlledState(
+    false,
+    visible,
+    onVisibleChange,
+  );
 
-  // Plugin to remove tabindex from tippy, to prevent focus ring from unintentionally showing.
-  const removeTabIndex = {
-    fn: () => ({
-      onCreate: (instance: Instance) => {
-        instance.popper.firstElementChild?.removeAttribute('tabindex');
-      },
+  const floating = useFloating({
+    placement,
+    open,
+    onOpenChange,
+    whileElementsMounted: (...args) => autoUpdate(...args, autoUpdateOptions),
+    middleware: [
+      middleware.offset !== undefined && offset(middleware.offset),
+      middleware.flip && flip(),
+      middleware.shift && shift(),
+      matchWidth &&
+        size({
+          apply: ({ rects }) => {
+            setReferenceWidth(rects.reference.width);
+          },
+        } as SizeOptions),
+      middleware.autoPlacement && autoPlacement(),
+      middleware.inline && inline(),
+      middleware.hide && hide(),
+    ].filter(Boolean),
+  });
+
+  const interactions = useInteractions([
+    useClick(floating.context, { enabled: !!trigger.click }),
+    useDismiss(floating.context, { outsidePress: closeOnOutsideClick }),
+    useHover(floating.context, {
+      enabled: !!trigger.hover,
+      delay: 100,
+      handleClose: safePolygon({ buffer: 1 }),
     }),
-  };
+    useFocus(floating.context, { enabled: !!trigger.focus }),
+    useRole(floating.context, { role: 'dialog', enabled: !!role }),
+  ]);
 
-  const computedProps: Partial<TippyProps> = {
-    allowHTML: true,
-    animation: false,
-    appendTo: (el) => themeInfo?.portalContainer || el.ownerDocument.body,
-    arrow: false,
-    duration: 0,
-    interactive: true,
-    role: '',
-    offset: [0, 0],
-    maxWidth: '',
-    zIndex: 99999,
-    ...props,
-    className: cx('iui-popover', props.className),
-    // add additional check for isDomAvailable when using in controlled mode,
-    // because rootRef is not available in first render
-    visible:
-      props.visible !== undefined ? props.visible && isDomAvailable : undefined,
-    plugins: [
-      lazyLoad,
-      removeTabIndex,
-      hideOnEscOrTab,
-      ...(props.plugins || []),
-    ],
-    popperOptions: {
-      strategy: 'fixed',
-      ...props.popperOptions,
-      modifiers: [
-        { name: 'flip' },
-        {
-          name: 'preventOverflow',
-          options: { padding: 0 },
+  const [referenceWidth, setReferenceWidth] = React.useState<number>();
+
+  const getFloatingProps = React.useCallback(
+    (userProps?: React.HTMLProps<HTMLElement>) =>
+      interactions.getFloatingProps({
+        ...userProps,
+        style: {
+          ...floating.floatingStyles,
+          zIndex: 9999,
+          ...(matchWidth && referenceWidth
+            ? {
+                minInlineSize: `${referenceWidth}px`,
+                maxInlineSize: `min(${referenceWidth * 2}px, 90vw)`,
```

Comment:
Is this to make sure the popover never becomes larger than the screen width? What does the `* 2` signify?

Reply:
yes we want it to never be larger than 2x the size of the trigger and also to never overflow the screen width

it's basically copied from before

https://github.com/iTwin/iTwinUI/blob/566257c666c26e357dc5be606bdc320db359e477/packages/itwinui-react/src/core/Select/Select.tsx#L383-L384